src/org/gk/qualityCheck/QACheckUtilities.java

```diff
@@ -26,4 +33,297 @@ public static GKInstance getLatestCuratorIEFromInstance(GKInstance instance) thr
+            }
+        }
+        return reactionsNotUsedForManualInference;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist in a ReactionlikeEvent's inputs, outputs, catalysts and regulations.
+     * @param reaction GKInstance -- ReactionlikeEvent that will be searched for PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntitys in ReactionlikeEvent
+     * @throws Exception -- Thrown by MySQLAdaptor
+     */
+    public static Set<GKInstance> findAllPhysicalEntitiesInReaction(GKInstance reaction) throws Exception {
+        Set<GKInstance> reactionPEs = new HashSet<>();
+        reactionPEs.addAll(findAllInputAndOutputPEs(reaction));
+        reactionPEs.addAll(findAllCatalystPEs(reaction));
+        reactionPEs.addAll(findAllRegulationPEs(reaction));
+        return reactionPEs;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist in a ReactionlikeEvent's inputs and outputs.
+     * @param reaction GKInstance -- ReactionlikeEvent that is being searched for PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntitys in ReactionlikeEvent's inputs and outputs.
+     * @throws Exception -- Thrown by MySQLAdaptor
+     */
+    public static Set<GKInstance> findAllInputAndOutputPEs(GKInstance reaction) throws Exception {
+        Set<GKInstance> inputOutputPEs = new HashSet<>();
+        for (String attribute : Arrays.asList(ReactomeJavaConstants.input, ReactomeJavaConstants.output)) {
+            for (GKInstance attributePE : (Collection<GKInstance>) reaction.getAttributeValuesList(attribute)) {
+                // QA checks calling these methods are concerned with PhysicalEntities that have a species attribute.
+                if (hasSpeciesAttribute(attributePE)) {
+                    inputOutputPEs.add(attributePE);
+                    inputOutputPEs.addAll(findAllPhysicalEntities(attributePE));
+                }
+            }
+        }
+        return inputOutputPEs;
+    }
+
+    /**
+     * Finds all PhysicalEntities that exist in a ReactionlikeEvent's catalystActivity. It checks the catalyst's
+     * activeUnit and physicalEntity attributes.
+     * @param reaction GKInstance -- ReactionlikeEvent that is being searched for PhysicalEntities.
+     * @return Set<GKInstance> -- All distinct PhysicalEntities in ReactionlikeEvent's catalystActivity.
+     * @throws Exception -- Thrown by MySQLAdaptor
+     */
+    public static Set<GKInstance> findAllCatalystPEs(GKInstance reaction) throws Exception {
+        Set<GKInstance> catalystPEs = new HashSet<>();
+        List<GKInstance> catalysts = reaction.getAttributeValuesList(ReactomeJavaConstants.catalystActivity);
+        for (GKInstance catalyst : catalysts) {
+            for (String attribute : Arrays.asList(ReactomeJavaConstants.activeUnit, ReactomeJavaConstants.physicalEntity)) {
+                for (GKInstance attributePE : (Collection<GKInstance>) catalyst.getAttributeValuesList(attribute)) {
```

Comment:
Completely a style choice, but an alternative to 3 nested for loops could be to delegate the inner two loops to call a method that takes a catalyst instance and returns its physical entities:

`private List<GKInstance> getCatalystPhysicalEntities(GKInstance catalyst)`

Reply:
It's a good suggestion, but I think having a `getCatalystPEs` within a method already called `findAllCatalystPEs` makes the overall Utilities class a little more muddled. I'll just add a comment that says that a catalysts PEs come from its ActiveUnit and PhysicalEntity slots